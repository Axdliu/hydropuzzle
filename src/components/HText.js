import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

class HText extends PureComponent {
  static propTypes = {
    type: PropTypes.string
  }

  static defaultProps = {
    style: {},
    type: 'default'
  }

  render () {
    const { style, type } = this.props;

    return (<Text {...this.props} style={[styles.common, styles[type], style]} />);
  }
}

export default HText;

const styles = EStyleSheet.create({
  common: {
    backgroundColor: 'transparent',
    fontSize: '1.1rem',
    color: 'black',
    fontWeight: '400'
  },
  default: {
    fontFamily: '$systemFont',
    lineHeight: '1.5rem'
  },
  mono: {
    fontFamily: '$monoFont',
    lineHeight: '1.5rem'
  },
  icon: {
    fontFamily: '$iconFont'
  }
});
