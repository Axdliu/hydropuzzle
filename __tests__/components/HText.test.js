import React from 'react';
import { shallow } from 'enzyme';

import HText from '@components/HText';

const text = 'Some text';

describe('HText', () => {
  const component = (<HText>{text}</HText>);

  it('renders correctly', () => {
    expect(shallow(component)).toMatchSnapshot();
  });

  it('contains text', () => {
    expect(shallow(component).contains(text)).toBe(true);
  });
});
