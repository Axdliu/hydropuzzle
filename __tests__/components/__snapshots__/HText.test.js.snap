// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HText renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "black",
        "fontSize": 17.05,
        "fontWeight": "400",
      },
      Object {
        "fontFamily": "System",
        "lineHeight": 23.25,
      },
      Object {},
    ]
  }
  type="default"
>
  Some text
</Text>
`;
